File: codewars/JavaScript/7kyu/Two fighters, one winner.js
function declareWinner(fighter1, fighter2, firstAttacker) {
  let secondAttacker = fighter2
  
  if(firstAttacker === fighter2.name) {
    firstAttacker = fighter2
    secondAttacker = fighter1
  } else {
    firstAttacker = fighter1
  }
  
  while(firstAttacker.health > 0 || secondAttacker.health > 0) {
    secondAttacker.health -= firstAttacker.damagePerAttack
    if(secondAttacker.health <= 0) return firstAttacker.name
    
    firstAttacker.health -= secondAttacker.damagePerAttack
    if(firstAttacker.health <= 0) return secondAttacker.name
  }
}

File: codewars/Python/7kyu/Two fighters, one winner.py
def declare_winner(fighter1, fighter2, first_attacker):
    if first_attacker == fighter2.name:
        fighter1, fighter2 = fighter2, fighter1
        
    while True:
        fighter2.health -= fighter1.damage_per_attack
        if(fighter2.health <= 0):
            return fighter1.name
        
        fighter1.health -= fighter2.damage_per_attack
        if(fighter1.health <= 0):
            return fighter2.name